Add info about files behavior (#4422)
vaadin/docs: articles/components/upload/index.adoc

endif::[]
--

> === Component state

Reviewer: ```suggestion
=== Component State
```

> In order to conserve resources the files themselves are not part of the component server state. This has significant implications in component behavior. Removing the Upload component from the layout will mean that files in browser memory will be lost. The same applies to refreshing of the page. Check more details about [file handling](https://vaadin.com/docs/latest/components/upload/file-handling) on a separate document page.

Reviewer: ```suggestion
To conserve resources, the files themselves aren't part of the component's server-side state. This has significant implications for component behavior. Removing the Upload component from the layout means that the files in the browser memory are lost. The same applies when reloading the page. Read more about <</components/upload/file-handling#, file handling>>.
```

> The UI including Upload component should take this into account. For example if refreshing of the page is expected, you can display a message to the user that files needs to be reselected if this was not the first attach.

Reviewer: ```suggestion
The user interface, including the Upload component, should take this into account. For example, if you expect that the user will refresh the page, you can display a message to the user that files needs to be selected again, if this wasn't the first time the page was rendered.
```
